app/src/main/cpp/skyline/services/codec/IHardwareOpusDecoder.h

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2021 Skyline Team and Contributors (https://github.com/skyline-emu/)
+
+#pragma once
+
+#include <opus.h>
+
+#include <common.h>
+#include <services/serviceman.h>
+
+namespace skyline::service::codec {
+    /**
+     * @brief This structure is located at the beginning of OpusDataIn for DecodeInterleaved*, with the actual opus packet following this
+     * @note These fields are big-endian
+     * @url https://github.com/switchbrew/libnx/blob/c5a9a909a91657a9818a3b7e18c9b91ff0cbb6e3/nx/include/switch/services/hwopus.h#L19
+     */
+    struct OpusDataHeader {
+        u32 size; //!< Size of the packet following this header
```

Comment:
Explicitly denote the endianess of this member, I'd even recommend making it `sizeBe` so that it wouldn't be used by mistake.